niworkflows/utils/testing.py

```diff
@@ -74,14 +74,15 @@ def generate_bids_skeleton(target_path, bids_config):
 
                 for bids_file in files:
                     metadata = bids_file.pop("metadata", None)
+                    extension = bids_file.pop("extension", ".nii.gz")
                     suffix = bids_file.pop("suffix")
                     entities = combine_entities(**bids_file)
-                    nii_file = modality_path / f"{bids_prefix}{entities}_{suffix}.nii.gz"
-                    nii_file.touch()
+                    out_file = modality_path / f"{bids_prefix}{entities}_{suffix}{extension}"
+                    out_file.touch()
```

Comment:
Nit-pick. `data_file` feels like a more descriptive name.

```suggestion
                    data_file = modality_path / f"{bids_prefix}{entities}_{suffix}{extension}"
                    data_file.touch()
```